runtime/src/chpl-gpu.c

```diff
@@ -700,9 +708,12 @@ void chpl_gpu_arg_reduce(void* _cfg, void* arg, size_t elem_size,
 
 void chpl_gpu_arg_host_register(void* _cfg, void* arg, size_t size) {
   kernel_cfg* cfg = (kernel_cfg*)_cfg;
-  cfg_add_direct_param(cfg, arg);
-  chpl_gpu_impl_host_register(*((void**)arg), sizeof(char)*8);
-  CHPL_GPU_DEBUG("\tAdded ref intent param (at %d): %p\n", cfg->cur_param,  arg);
+
+  void *dev_arg = chpl_gpu_impl_host_register(*((void**)arg), size);
+  *(cfg->host_registered_vars[cfg->cur_host_registered_var]) = dev_arg;
+  cfg_add_direct_param(cfg, cfg->host_registered_vars[cfg->cur_host_registered_var]);
```

Comment:
The parameters we pass to the kernel launchers are addresses of things that we want to pass as arguments. `dev_arg` is the thing we want to pass, so we actually have to pass its address. Here, you made `cfg->host_registered_vars` an array that can store such addresses. You couldn't have passed `&dev_arg` as it'll go out of scope.

All this is to say that this makes sense but annoyingly convoluted. We have a similar pattern elsewhere in the runtime I believe. Maybe the `offload` version of arg passing?